invenio_userprofiles/views.py

```diff
@@ -99,20 +101,32 @@ def profile():
     preferences_form = PreferencesForm(
         formdata=None, obj=current_user, prefix="preferences")
 
-    # Process forms
+    # Pick form
     is_read_only = current_app.config.get("USERPROFILES_READ_ONLY", False)
-    form = request.form.get('submit', None)
-    if form == 'profile' and not is_read_only:
-        handle_profile_form(profile_form)
-    elif form == 'verification':
-        handle_verification_form(verification_form)
-    elif form == 'preferences' and not is_read_only:
-        handle_preferences_form(preferences_form)
+    form_name = request.form.get('submit', None)
+    if form_name == 'profile' and not is_read_only:
+        handle_form = handle_profile_form
+        form = profile_form
+    elif form_name == 'verification':
+        handle_form = handle_verification_form
+        form = verification_form
+    elif form_name == 'preferences':
+        handle_form = handle_preferences_form
+        form = preferences_form
+    else:
+        form = None
+
+    # Process form
+    if form:
+        form.process(formdata=request.form)
+        if form.validate_on_submit():
+            handle_form(form)
+            return redirect(url_for(".profile"))  # this endpoint
```

Comment:
```suggestion
            return redirect(url_for(".profile"), code=HTTP_STATUS_CODES[303])  # this endpoint
```
due to 303 handling this exact case, see https://serverfault.com/a/391182

Reply:
Thanks for that, I didn't know about 303! Here is what I understood: using the 303 HTTP return code tells the browser to redirect using a GET, so we are sure a POST is not made on the Location. Although most browsers didn't do a POST on the Location of a 302, this makes it explicit.